apps/prairielearn/src/sync/syncFromDisk.ts

```diff
@@ -43,20 +51,72 @@ export async function checkSharingConfigurationValid(
   courseData: courseDB.CourseData,
   logger: Logger,
 ): Promise<boolean> {
-  if (config.checkSharingOnSync) {
-    // TODO: also check if questions were un-shared in the JSON or if any
-    // sharing sets were deleted
-    const invalidRenames = await getInvalidRenames(courseId, courseData);
-    if (invalidRenames.length > 0) {
-      logger.info(
-        chalk.red(
-          `✖ Course sync completely failed. The following questions are shared and cannot be renamed or deleted: ${invalidRenames.join(', ')}`,
-        ),
-      );
-      return false;
-    }
+  if (!config.checkSharingOnSync) {
+    return true;
+  }
+
+  let sharingConfigurationValid = true;
+
+  const sharedQuestions = await selectSharedQuestions(courseId);
+  const invalidRenames = getInvalidRenames(sharedQuestions, courseData);
+  if (invalidRenames.length > 0) {
+    logger.info(
+      chalk.red(
```

Comment:
Nit: our job loggers support `logger.error` which will automatically log in red. If you want to take advantage of this, you'll need to update the `Logger` interface above and update any callsites that aren't providing a logger with that function (most likely just tests).

Reply:
done. 

Its funny I didn't realize that, because I was just copying the way things were done in some other code in this file. I guess we should update that to do the same things as well (at some point, not now).